### PR TITLE
Random Doubles Updates

### PR DESCRIPTION
**Pull request by ACakeWearingAHat**
**DO NOT PATCH THIS IN UNTIL ALL RANDOM BATTLES TEAM TOUR 4 WEEK 6 DOUBLES GAMES ARE COMPLETE PLEASE** (feel free to still review, though)

**Patch notes (2/9/21):**
-Prevent Virizion with 1 attack by preventing Coaching from generating with any other setup move.
-Prevent First Impression from generating with any setup move, for Sirfetch'd.
-Reject Protect if Belly Drum is in the movepool but hasn't been rolled, in an attempt to prevent Belly Drum-less Eiscue.
-Cofagrigus, Melmetal: Prevent generation of Defense-raising moves without Body Press by making sure Body Press doesn't get rejected in favor of STAB (Not yet implemented; planned by the time update is done)
-Glastrier: Icicle Crash over Avalanche
-Keldeo: Muddy Water over Hydro Pump and Scald, added Air Slash.
-Regieleki: Electroweb over Eerie Impulse
-Various level balancing. Pokemon above 56.00% win rate nerfed. Pokemon below 46.00% win rate buffed. Highlights: Moltres-Galar, Regidrago, Calyrex-Shadow, Zacian-C down again, Maractus, Toxapex, Shuckle, etc. up. See pull request for full list of level changes.

Gen 7 Random Doubles: Remove Protect from Typhlosion to prevent Choice Scarf Protect.

### CODE REVIEW

File: data/random-teams.ts [hunk 1]
```diff
@@ -733,6 +733,9 @@ export class RandomTeams {
 					if ((hasMove['substitute'] && !hasType['Fighting']) || hasMove['toxic'] && movePool.includes('substitute')) rejected = true;
 					if (moveid === 'closecombat' && (hasMove['highjumpkick'] || movePool.includes('highjumpkick')) && !counter.setupType) rejected = true;
 					break;
+				case 'coaching':
+					if (counter.setupType) rejected = true;
+					break;
```
Comment (TheImmortal) [inline data/random-teams.ts:738]: There's no need to make an ew case for this. Just add coaching on line 723.

Comment (ACakeWearingAHat) [reply data/random-teams.ts:738]: u rite

**Review by TheImmortal — APPROVED** [review]

File: data/random-teams.ts [hunk 2]
```diff
@@ -752,7 +752,7 @@ export class RandomTeams {
 					if ((counter.setupType && !hasMove['wish'] && !isDoubles) || hasMove['rest'] && hasMove['sleeptalk']) rejected = true;
 					if (counter.Status < 2 && !hasAbility['Hunger Switch'] && !hasAbility['Speed Boost'] && !isDoubles) rejected = true;
 					if (movePool.includes('leechseed') || movePool.includes('toxic') && !hasMove['wish']) rejected = true;
-					if (isDoubles && (movePool.includes('fakeout') || movePool.includes('shellsmash') || movePool.includes('spore') || hasMove['tailwind'] || hasMove['waterspout'])) rejected = true;
+					if (isDoubles && (movePool.includes('bellydrum') || movePool.includes('fakeout') || movePool.includes('shellsmash') || movePool.includes('spore') || hasMove['tailwind'] || hasMove['waterspout'])) rejected = true;
```
Comment (AnnikaCodes) [inline data/random-teams.ts:755]:
```suggestion
					if (isDoubles && (['bellydrum', 'fakeout', 'shellsmash', 'spore').some(m => movePool.includes(m)) || hasMove['tailwind'] || hasMove['waterspout'])) rejected = true;
```

**Review by AnnikaCodes — APPROVED** [review]
This looks good, thanks! Let me know when the relevant RBTT games are over and I'll merge this.

File: data/random-teams.ts [hunk 3]
```diff
@@ -752,7 +752,7 @@ export class RandomTeams {
 					if ((counter.setupType && !hasMove['wish'] && !isDoubles) || hasMove['rest'] && hasMove['sleeptalk']) rejected = true;
 					if (counter.Status < 2 && !hasAbility['Hunger Switch'] && !hasAbility['Speed Boost'] && !isDoubles) rejected = true;
 					if (movePool.includes('leechseed') || movePool.includes('toxic') && !hasMove['wish']) rejected = true;
-					if (isDoubles && (movePool.includes('fakeout') || movePool.includes('shellsmash') || movePool.includes('spore') || hasMove['tailwind'] || hasMove['waterspout'])) rejected = true;
+					if (isDoubles && (('bellydrum', 'fakeout', 'shellsmash', 'spore').some(m => movePool.includes(m)) || hasMove['tailwind'] || hasMove['waterspout'])) rejected = true;
```
Comment (AnnikaCodes) [inline data/random-teams.ts:755]:
```suggestion
					if (isDoubles && (['bellydrum', 'fakeout', 'shellsmash', 'spore'].some(m => movePool.includes(m) || hasMove['tailwind'] || hasMove['waterspout'])) rejected = true;
```

File: data/random-teams.ts [hunk 4]
```diff
@@ -752,7 +752,7 @@ export class RandomTeams {
 					if ((counter.setupType && !hasMove['wish'] && !isDoubles) || hasMove['rest'] && hasMove['sleeptalk']) rejected = true;
 					if (counter.Status < 2 && !hasAbility['Hunger Switch'] && !hasAbility['Speed Boost'] && !isDoubles) rejected = true;
 					if (movePool.includes('leechseed') || movePool.includes('toxic') && !hasMove['wish']) rejected = true;
-					if (isDoubles && (movePool.includes('fakeout') || movePool.includes('shellsmash') || movePool.includes('spore') || hasMove['tailwind'] || hasMove['waterspout'])) rejected = true;
+					if (isDoubles && (['bellydrum', 'fakeout', 'shellsmash', 'spore'].some(m => movePool.includes(m) || hasMove['tailwind'] || hasMove['waterspout'])) rejected = true;
```
Comment (AnnikaCodes) [inline data/random-teams.ts:755]:
```suggestion
					if (isDoubles && (['bellydrum', 'fakeout', 'shellsmash', 'spore'].some(m => movePool.includes(m)) || hasMove['tailwind'] || hasMove['waterspout'])) rejected = true;
```

**Pull request closed without merging**